Add monthly workflow to update dependencies

bgilbert opened the pull request:
Disable Dependabot; it's too noisy.  Instead, run `cargo update` once a month and submit a PR.

This won't notice `nix` minor version bumps, unfortunately, and neither does deps.rs (I guess because it's an arch-specific dependency?).  And we probably don't want to declare compatibility with `nix` `>= 0.22, < 1`, since we can't promise that.  I don't have a suggestion here, but I'm okay with leaving that problem unresolved for now.

Sample PR in https://github.com/bgilbert/gptman/pull/4.

cecton reviewed (approved):
Nice that's fantastic really!

I'm deeply sorry that I dragged you into all that maintenance stuff for 1.0. I really not meant to do this :weary:

bgilbert commented:
No worries!  I actually kind of enjoy this sort of maintenance work, and I think the infrastructure is in a bit better shape now.  Thanks for handling all the reviews, cross-linked PRs, email, and churn.  :sweat_smile:

Rebased on `main` and dropped the `nix` update in favor of doing it separately in #91.

cecton commented:
> No worries! I actually kind of enjoy this sort of maintenance work, and I think the infrastructure is in a bit better shape now. Thanks for handling all the reviews, cross-linked PRs, email, and churn. sweat_smile
> 
> Rebased on `main` and dropped the `nix` update in favor of doing it separately in #91.

You should keep that as a secret and never tell anyone, especially at work, or you will be assigned to devops work for the rest of your life xD

(You can tell it's based on real life experience)

bgilbert commented:
Yeah, I'm pretty sure they know.  :smirk:
